src/electrum.rs

```diff
@@ -275,6 +275,88 @@ impl Rpc {
         Ok(status)
     }
 
+    fn outpoint_subscribe(&self, (txid, vout): (Txid, u32)) -> Result<Value> {
+        let funding = OutPoint { txid, vout };
+
+        let funding_blockhash = self.tracker.get_blockhash_by_txid(funding.txid);
+        let spending_blockhash = self.tracker.get_blockhash_spending_by_outpoint(funding);
+
+        let mut inputs_funding_confirmed = true;
+
+        let funding_tx = self.daemon.get_transaction(&funding.txid, funding_blockhash);
+
+        if funding_tx.is_err() {
+            return Ok(json!({}));
+        }
+
+        let funding_inputs = &funding_tx.as_ref().unwrap().input;
+        let outpoint_script = &funding_tx.as_ref().unwrap().output.get(vout as usize).unwrap().script_pubkey;
+
+        if funding_blockhash.is_none() && funding_inputs.iter().any(|txi| self.tracker.get_blockhash_by_txid(txi.previous_output.txid).is_none()) {
```

Comment:
I didn't know how to present that more clearly in the conde but `inputs_funding_confirmed` only matters when the funding transaction is in the mempool. 

This is not critical but according to [specs](https://github.com/SomberNight/electrumx/blob/202012_protocol_1_5/docs/protocol-methods.rst#blockchainoutpointsubscribe) the method must return funding height of 0 if inputs are all confirmed  else -1. `inputs_funding_confirmed` just check that and is then used to [return the correct output](https://github.com/romanz/electrs/pull/446/files#diff-ede2fa6030c1edb89d5de41ac20473c31669e46197974a46108a9bff88eebfaeR324) for the funding height. 

`funding_blockhash.is_none()` test is just to avoid iterating over fundings inputs when it is confirmed (since inputs are then necessarly confirmed in that case as you said).

Maybe I should put that in the if branch when spending transaction is not confirmed. What do you think ?

Reply:
Ah, I see now, I misunderstood what it's doing. Perhaps `funding_inputs_confirmed` would be a clearer name?

Reply:
Done !